docs/rules/no-eval.md

```diff
@@ -30,8 +52,86 @@ The following patterns are not considered problems:
 var obj = { x: "foo" },
     key = "x",
     value = obj[key];
+
+class A {
+    foo() {
+        // This is a user-defined method.
+        this.eval("var a = 0");
+    }
+
+    eval() {
+    }
+}
+```
+
+### Options
+
+This rule has an option to allow calls of indirect `eval`.
+Calls of indirect `eval` are dangerous less than calls of direct `eval` relatively.
```

Comment:
"Indirect calls to eval are less dangerous than direct calls to eval because they cannot dynamically change the scope. Because of this, they also will not negatively impact performance to the degree of direct eval."